packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -561,13 +542,16 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
     }
 
     public async delete(uri: vscode.Uri, _options: { readonly recursive: boolean }): Promise<void> {
-        const entry = this.lookup(uri, false);
+        const entry = this.lookup(uri, false) as DsEntry | PdsEntry;
         const parent = this._lookupParentDirectory(uri);
```

Comment:
It would seems that the `this.lookup` method is called with the same `uri` many times (twice in this scenario)

I see that the lookup operation is not very complex, but curious if there is a way to reduce the number of times we have to perform those activities 😅 

Not requesting changes here though 😋

Reply:
Lookup is called twice here but with different URIs:

- `uri` parameter is passed to `this.lookup` to get its entry
- `uri` parameter is passed to `this.lookupParentDirectory`, which builds a URI "one level upward" and then calls `this.lookup` on the new URI that was built

Reply:
That makes sense.
It's nice to see that it just works by doing `..` (at the end) to traverse upward 🙏🏽 

Thanks for explaining 🥳